develop/dev-guide-third-party-tools-compatibility.md

```diff
@@ -0,0 +1,158 @@
+---
+
+# Known Incompatibility Issues with Third-Party Tools
+
+> **Note:**
+>
+> The [Unsupported features](/mysql-compatibility.md#unsupported-features) lists features that are not supported in TiDB, including:
+>
+> - Stored procedures and functions
+> - Triggers
+> - Events
+> - User-defined functions
+> - `FOREIGN KEY` constraints
+> - `SPATIAL` functions, data types and indexes
+> - `XA` syntax
+>
+> The preceding unsupported features are considered expected behavior and are not listed in this document. For more details, see [MySQL Compatibility](/mysql-compatibility.md).
+
+## General incompatibility
+
+### `SELECT CONNECTION_ID()` returns a 64-bit integer in TiDB
+
+**Description**
+
+The `SELECT CONNECTION_ID()` function returns a 64-bit integer in TiDB, such as `2199023260887`, while it returns a 32-bit integer in MySQL, such as `391650`.
+
+**Way to avoid**
+
+In a TiDB application, to avoid data overflow, you should use a 64-bit integer or string type to store the result of `SELECT CONNECTION_ID()`. For example, you can use `Long` or `String` in Java and use `string` in JavaScript or TypeScript.
+
+### TiDB does not maintain `Com_*` counters
+
+**Description**
+
+MySQL maintains a series of [server status variables starting with `Com_`](https://dev.mysql.com/doc/refman/8.0/en/server-status-variables.html#statvar_Com_xxx) to keep track of the total number of operations you have performed on the database. For example, `Com_select` records the total number of `SELECT` statements initiated by MySQL since it was last started (even if the statements were not queried successfully). TiDB does not maintain these variables. You can use the statement `SHOW GLOBAL STATUS LIKE 'Com_%'` to see the difference between TiDB and MySQL.
+
+**Way to avoid**
+
+Please do not use these variables. One of the common usage scenarios is monitoring. TiDB is well observable and does not require querying from server status variables. For custom monitoring tools, refer to [TiDB Monitoring Framework Overview](/tidb-monitoring-framework.md).
+
+### TiDB distinguishes between `TIMESTAMP` and `DATETIME` in error messages
+
+**Description**
+
+TiDB error messages distinguish between `TIMESTAMP` and `DATETIME`, while MySQL does not, and returns them all as `DATETIME`. That is, MySQL incorrectly converts `TIMESTAMP` type error messages to `DATETIME` type.
+
+**Way to avoid**
+
+Do not use the error messages for string matching, but use [Error Codes](/error-codes.md) for troubleshooting.
+
+### TiDB does not support the `CHECK TABLE` statement
+
+**Description**
+
+The `CHECK TABLE` statement is not supported in TiDB. To check the consistency of data and corresponding indexes, you can use the [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) statement in TiDB.
+
+**Way to avoid**
+
+Use [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) instead.
```

Comment:
```suggestion
The `CHECK TABLE` statement is not supported in TiDB. 

**Way to avoid**

To check the consistency of data and corresponding indexes, you can use the [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) statement in TiDB.
```